workers/api-server/src/http/admin.rs

```diff
@@ -105,21 +121,82 @@ impl TypedHandler for AdminOpenOrdersHandler {
         _headers: HeaderMap,
         _req: Self::Request,
         _params: UrlParams,
-        _query_params: QueryParams,
+        query_params: QueryParams,
     ) -> Result<Self::Response, ApiServerError> {
-        let order_ids = self.state.get_locally_matchable_orders().await?;
+        let order_ids =
+            if let Some(matching_pool) = parse_matching_pool_from_query_params(&query_params) {
+                // Check that the matching pool exists
+                if !self.state.matching_pool_exists(matching_pool.clone()).await? {
+                    return Err(not_found(ERR_NO_MATCHING_POOL));
+                }
+
+                self.state.get_locally_matchable_orders_in_matching_pool(matching_pool).await?
+            } else {
+                self.state.get_locally_matchable_orders().await?
+            };
+
         let mut orders = Vec::new();
         for id in order_ids.into_iter() {
             let order = self.state.get_order_metadata(&id).await?;
             if let Some(meta) = order {
-                orders.push(meta);
+                let (fillable, price) = get_fillable_amount_and_price(
+                    &meta,
+                    &self.state,
+                    &self.price_reporter_job_queue,
+                )
+                .await?;
+                orders.push(OpenOrder { order: meta, fillable, price })
             }
         }
 
         Ok(OpenOrdersResponse { orders })
     }
 }
 
+/// Get the fillable amount of an order using the underlying wallet's balances,
+/// & potentially the price of the base asset
+async fn get_fillable_amount_and_price(
```

Comment:
nit: can we move this to a helpers section? it's nice to have all the route handlers next to one another